yt/frontends/cf_radial/data_structures.py

```diff
@@ -0,0 +1,329 @@
+                            "Ignoring provided grid parameters because %s exists.",
+                            storage_filename,
+                        )
+                        mylog.warning(
+                            "To re-grid, either provide a unique storage_filename or set "
+                            "storage_overwrite to True to overwrite %s.",
+                            storage_filename,
+                        )
+                        regrid = False
+                    else:
+                        mylog.info(
+                            "loading existing re-gridded file: %s", storage_filename
+                        )
+                        regrid = False
+
+                if regrid:
+                    mylog.info("Building cfradial grid")
+                    from yt.utilities.on_demand_imports import _pyart as pyart
+
+                    radar = pyart.io.read_cfradial(filename)
+
+                    grid_shape, grid_limits = self._validate_grid(
+                        radar, grid_shape, grid_limit_x, grid_limit_y, grid_limit_z
+                    )
+
+                    # grid_shape and grid_limits are in (z, y, x) order.
+                    self.grid_shape = grid_shape
+                    self.grid_limits = grid_limits
+                    mylog.info("Calling pyart.map.grid_from_radars ... ")
+                    # this is fairly slow
+                    grid = pyart.map.grid_from_radars(
+                        (radar,),
+                        grid_shape=self.grid_shape,
+                        grid_limits=self.grid_limits,
+                    )
+                    mylog.info(
+                        "Successfully built cfradial grid, writing to %s",
+                        storage_filename,
+                    )
+                    mylog.info(
+                        "Subsequent loads of %s will load the gridded file by default",
+                        filename,
+                    )
+                    grid.write(storage_filename)
+
+                filename = storage_filename
+                self.storage_filename = storage_filename
+
+        super().__init__(filename, dataset_type, units_override=units_override)
+        self.refine_by = 2  # refinement factor between a grid and its subgrid
+
+    @contextlib.contextmanager
+    def _handle(self, filename: Optional[str] = None):
+        if filename is None:
+            if hasattr(self, "filename"):
+                filename = self.filename
+            else:
+                raise RuntimeError("Dataset has no filename yet.")
+
+        with xr.open_dataset(filename) as xrds:
+            yield xrds
+
+    def _validate_grid(
+        self,
+        radar,
+        grid_shape: Optional[Tuple[int, int, int]] = None,
+        grid_limit_x: Optional[Tuple[float, float]] = None,
+        grid_limit_y: Optional[Tuple[float, float]] = None,
+        grid_limit_z: Optional[Tuple[float, float]] = None,
+    ) -> Tuple[Tuple, Tuple]:
```

Comment:
Note that this is implicitly equivalent to `Tuple[Tuple[Any, ...], Tuple[Any, ...]]` and I think that at least one of the errors reported by mypy is a direct consequence of this type being too loose

Reply:
Ok! Ya, this is causing issues... but I finally got mypy working again locally after some issues with my environment, so next push should hopefully resolve the type checking. Will wait to see if all the builds pass though... I **think** I fixed the remaining issues that were causing those to fail.

Reply:
Latest commits split up this function and it made the typing much easier to get passing mypy. I'll go ahead and resolve the conversation since this section of code is outdated now, but feel free to suggest more typing improvements if you have ideas.